docs/packages/markdown/loader.js

```diff
@@ -187,27 +190,51 @@ module.exports = async function demoLoader() {
     }),
   );
 
-  /**
-   * @param {string} moduleID
-   */
-  function getDemoIdentifier(moduleID) {
-    return moduleIDToJSIdentifier(moduleID);
-  }
+  const componentNames = Array.from(
+    new Set(
+      docs.en.rendered
+        .filter((markdownOrComponentConfig) => {
+          return (
+            typeof markdownOrComponentConfig !== 'string' && markdownOrComponentConfig.component
+          );
+        })
+        .map((componentConfig) => {
+          return componentConfig.component;
+        }),
+    ),
+  );
+
+  componentNames.forEach((componentName) => {
+    const moduleID = path.join(this.rootContext, 'src', componentName);
+
+    components[moduleID] = componentName;
+    componentModuleIDs.add(moduleID);
+  });
 
   const transformed = `
     ${Array.from(demoModuleIDs)
       .map((moduleID) => {
-        return `import ${getDemoIdentifier(moduleID)} from '${moduleID}';`;
+        return `import ${moduleIDToJSIdentifier(moduleID)} from '${moduleID}';`;
+      })
+      .join('\n')}
+    ${Array.from(componentModuleIDs)
+      .map((moduleID) => {
+        return `import ${moduleIDToJSIdentifier(moduleID)} from '${moduleID}';`;
       })
       .join('\n')}
 
     export const docs = ${JSON.stringify(docs, null, 2)};
     export const demos = ${JSON.stringify(demos, null, 2)};
     export const demoComponents = {${Array.from(demoModuleIDs)
       .map((moduleID) => {
-        return `${JSON.stringify(moduleID)}: ${getDemoIdentifier(moduleID)},`;
+        return `${JSON.stringify(moduleID)}: ${moduleIDToJSIdentifier(moduleID)},`;
       })
       .join('\n')}};
+        export const componentComponents = {${Array.from(componentModuleIDs)
```

Comment:
```suggestion
        export const srcComponents = {${Array.from(componentModuleIDs)
```
I rather go with `srcComponents` to be more explicit because the module id above is tight to path `docs/src/*`.

Reply:
@flaviendelangle is the name `srcComponents` sounds better to you?

Reply:
I'm fine with it yes :+1:

Reply:
Done